x-pack/plugin/async-search/src/main/java/org/elasticsearch/xpack/search/TransportSubmitAsyncSearchAction.java

```diff
@@ -185,11 +188,45 @@ private void onFinalResponse(AsyncSearchTask searchTask,
                             cause instanceof VersionConflictEngineException == false) {
                         logger.error(() -> new ParameterizedMessage("failed to store async-search [{}]",
                             searchTask.getExecutionId().getEncoded()), exc);
+                        updateStoredResponseWithFailure(searchTask.getExecutionId().getDocId(), response, exc);
```

Comment:
I think we should only call `unregisterTaskAndMoveOn` after `updateStoredResponseWithFailure` has completed.

Reply:
@dnhatn Thanks for additional comments, addressed in ecf614d35339fe9f3388c6f82ca0d79d089a51ab